### ISSUE TITLE
[ENH] improve test framework for metrics in v1.

### ISSUE THREAD

**Issue by PranavBhatP** [jdb78/pytorch-forecasting#1904]
**Is your feature request related to a problem? Please describe.**
Based on findings of the draft design [document](https://hackmd.io/DL0_NjbERIy65laob22GlA?edit) and #1900 , there is a need to test all the existing functionality of metrics in v1. Currently, there is only a single `test_metrics.py` file which tests the existing metrics and there is a scope for improving the design of the test framework, before proceeding with any changes related to standardization of output format of models.

**Describe the solution you'd like**
Implement a testing framework to loop through all the loss functions and test for basic must-have functionalities and attributes. Following this, individual features can be tested to improve coverage and test correct usage as outlined in vignettes.

**Comment by PranavBhatP** [comment]
@phoeenniixx @fkiraly, I had a question regarding the fixture generator to be used for testing metrics. Are we allowed to alter the `fixture_sequence` inside the metric fixture generator class inheriting from `BaseFixtureGenerator`? I don't think `trainer_kwargs` would be applicable for metrics.

Ref: https://github.com/sktime/pytorch-forecasting/blob/main/pytorch_forecasting/tests/test_all_estimators.py#L110

**Comment by fkiraly** [comment]
> Are we allowed to alter the fixture_sequence inside the metric fixture generator class inheriting from BaseFixtureGenerator?

Yes, all params can be overridden if you want. The key issue may be is that metrics inherit from `LightningMetric` and not from `BaseObject`, different options are discussed in your PR #1907
